pkg/restore/restore.go

```diff
@@ -363,6 +419,41 @@ func (ctx *context) restoreFromDir(dir string) (api.RestoreResult, api.RestoreRe
 	return warnings, errs
 }
 
+// getNamespace returns a namespace API object that we should attempt to
+// create before restoring anything into it. It will come from the backup
+// tarball if it exists, else will be a new one. If from the tarball, it
+// will retain its labels, annotations, and spec.
+func getNamespace(path, remappedName string) *v1.Namespace {
+	var nsBytes []byte
+	var err error
+
+	if nsBytes, err = ioutil.ReadFile(path); err != nil {
+		return &v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: remappedName,
+			},
+		}
+	}
+
+	var backupNS v1.Namespace
+	if err := json.Unmarshal(nsBytes, &backupNS); err != nil {
+		return &v1.Namespace{
```

Comment:
Should we log here? Or return an error? Presumably it means the file is corrupt?

Reply:
I think logging a warning is reasonable -- I don't really see a reason to error out though. Do you?

Reply:
It just means that you have to go digging for why your labels/annotations weren't restored properly. I think logging is a good enough place to start.